Sim/SimPapas/src/PapasAlg.h

```diff
@@ -35,16 +41,24 @@ class PapasAlg : public GaudiAlgorithm {
   virtual StatusCode finalize();
 
 private:
+  std::shared_ptr<papas::Detector> m_spDetector;  ///< papas detector
   /// Stores the linkages between papas objects (eg clusters, tracks, particles)
   papas::Nodes m_history;
-  std::vector<IPapasTool*> m_tools;      ///<vector of tools to be run
-  std::vector<std::string> m_toolNames;  ///< names of tools to be run
-  long m_eventno;                        ///< the papas Event number, incremented for each event processed
+  std::vector<IPapasTool*> m_tools;       ///< vector of tools to be run
+  IPapasImportTool* m_importTool;         ///< import tool to be run
+  IPapasExportTool* m_exportTool;         ///< export tool to be run
+  std::vector<std::string> m_toolNames;   ///< names of tools to be run
+  std::string m_importToolName;           ///< name of import tool to be run
+  std::string m_exportToolName;           ///< name of export tool to be run
+  std::unordered_map<papas::Identifier, int> m_importParticleLinks; ///< map to contain links between GenParticles and SimParticles
+  long m_eventno;                         ///< the papas Event number, incremented for each event processed
```

Comment:
Is that really necessary? could we use the gaudi event number?

Reply:
Now done.